media/js/firefox/concerts.js

```diff
@@ -1,6 +1,173 @@
+        };
+
+        xhr.open('GET', '/country-code.json');
+        // must come after open call above for IE 10 & 11
+        xhr.timeout = 2000;
+        xhr.send();
+    };
+
+    ConcertPage.hasGeoOverride = function(location) {
+        var loc = location || window.location.search;
+        if (loc.indexOf('geo=') !== -1) {
+            var urlRe = /geo=([a-z]{2})/i;
+            var match = urlRe.exec(loc);
+            if (match) {
+                return match[1].toLowerCase();
+            }
+            return false;
+        }
+        return false;
+    };
+
+    ConcertPage.verifyLocation = function(location) {
+        if (location) {
+            return location !== ConcertPage.US_COUNTRY_CODE;
+        }
+
+        return false;
+    };
+
+    ConcertPage.onRequestComplete = function(data) {
+        var country = typeof data === 'string' ? data : 'none';
+
+        clearTimeout(_geoTimeout);
+
+        if (!_requestComplete) {
+            _requestComplete = true;
+
+            // Set a cookie so we don't have to query location on repeated page loads.
+            ConcertPage.setCookie(country);
+
+            // Update page content based on location.
+            ConcertPage.updatePageContent();
+        }
+    };
+
+    ConcertPage.updatePageContent = function() {
+        if (ConcertPage.shouldShowConcert() === 'false') {
+            ConcertPage.showExcludedContent();
+        } else {
+            ConcertPage.showConcertContent();
+        }
+    };
+
+    ConcertPage.showConcertContent = function() {
+        // Add styling hook for us-specific CSS.
+        document.body.classList.add('state-in-us');
+    };
+
+    ConcertPage.showExcludedContent = function() {
+        // Add styling hook for excluded-specific CSS.
+        document.body.classList.add('state-not-us');
+    };
+
+    ConcertPage.shouldShowConcert = function() {
+        // Is user in the US?
+        return ConcertPage.verifyLocation(ConcertPage.getCookie(ConcertPage.COOKIE_ID));
+    };
+
+    ConcertPage.cookieExpiresDate = function(date) {
+        var d = date || new Date();
+        d.setTime(d.getTime() + (ConcertPage.COOKIE_EXPIRATION_DAYS * 24 * 60 * 60 * 1000));
+        return d.toUTCString();
+    };
+
+    ConcertPage.setCookie = function(country) {
+        Mozilla.Cookies.setItem(ConcertPage.COOKIE_ID, country, ConcertPage.cookieExpiresDate());
+    };
+
+    ConcertPage.getCookie = function(id) {
+        return Mozilla.Cookies.getItem(id);
+    };
+
+    ConcertPage.hasCookie = function() {
+        return Mozilla.Cookies.hasItem(ConcertPage.COOKIE_ID);
+    };
+
+    ConcertPage.init = function() {
+        ConcertPage.passThroughParams();
+
+        var cookiesEnabled = typeof Mozilla.Cookies !== 'undefined' || Mozilla.Cookies.enabled();
+        var override = ConcertPage.hasGeoOverride();
+
+        // only show excluded content if cookies are enabled (let LiveNation deal with it)
+        if (cookiesEnabled) {
```

Comment:
Not a blocker but we could still use the geofencing if cookies are disabled. This should be updated *after* we get this into production.

Reply:
Good point... this was cribbed from the Yandex page, which I think had a more strict cookie requirement because it's doing more than just geofencing.